hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/compact/CompactSerializationTest.java

```diff
@@ -430,6 +439,125 @@ public void testReflectiveSerializer_withFieldsWithOtherSerializationMechanisms_
         assertTrue(data.isCompact());
     }
 
+    @Test
+    public void testCompactWriterThrowsExceptionWhileWritingWhenSchemaDoesNotHaveThatField() {
+        thrown.expect(HazelcastSerializationException.class);
+        thrown.expectMessage("Failed to serialize");
+        thrown.expectCause(new RootCauseMatcher(HazelcastSerializationException.class, "Invalid field name"));
+
+        FooBarSerializer serializer = spy(new FooBarSerializer());
+        FooBar fooBar = new FooBar("foo", 42L);
+
+        SchemaWriter schemaWriter = new SchemaWriter(serializer.getTypeName());
+        serializer.write(schemaWriter, fooBar);
+        Schema schema = schemaWriter.build();
+
+        SchemaService schemaService = CompactTestUtil.createInMemorySchemaService();
+        schemaService.put(schema);
+        CompactSerializationConfig compactSerializationConfig = new CompactSerializationConfig();
+        compactSerializationConfig.addSerializer(serializer);
+        SerializationService serializationService = new DefaultSerializationServiceBuilder()
+                .setSchemaService(schemaService)
+                .setConfig(new SerializationConfig().setCompactSerializationConfig(compactSerializationConfig))
+                .build();
+
+        // Add correct schema to classToSchemaMap
+        serializationService.toData(fooBar);
```

Comment:
Do we need to add schema to the schema service above? This line would do it as well, right?

Reply:
Correct, removing